install/kubernetes/helm/istio/values.yaml

```diff
@@ -151,6 +151,16 @@ global:
   # will be exposed on an internal gateway
   meshExpansionILB: false
 
+  # Set to true to connect two kubernetes clusters using a LB gateway as the
+  # only entry point into the cluster (instead of requiring pod to pod
+  # connectivity across two clusters). Note that for this system to work,
+  # service objects from remote clusters have to be replicated to local
+  # cluster (without the pod selectors). In addition, endpoints have to be
```

Comment:
Did you try this with 'bring your own name' - i.e. use a real DNS to define the name ? I think this could even work if the DNS entry is a CNAME to the ingress gateway.

But I agree the simplest mechanism is to replicate headless services with explicit IP.

Reply:
Kubernetes will not forward queries for *.local to any external DNS. It might work if the clusters have different DNS suffixes but for that, I already have a coreDNS plugin.
Secondly, this is not a headless service. Its almost the same service, without selectors, (kube still assigns a cluster IP to the service in the local cluster). We create an endpoint object that has the IPs of the ingress gateway. We can't create a service with an IP thats not part of the cluster CIDR. The endpoint is the only way to go..
A variant of this approach (https://github.com/vmware/k8s-endpoints-sync-controller) where instead of replicating hte actual pod IPs, we stick in the gateway's IP for the endpoints.